packages/sil-governance/src/store/proposals/actions.ts

```diff
@@ -74,3 +79,7 @@ export const replaceProposalsAction = (proposals: ProposalsState): ReplacePropos
   type: "@@proposals/REPLACE",
   payload: proposals,
 });
+
+export const refreshProposalsAction = (): RefreshProposalsActionType => ({
+  type: "@@proposals/REFRESH",
+});
```

Comment:
I don't think this is an "action" is the sense of redux (https://redux.js.org/basics/actions), as it does not manipulate the store. It is an instruction to potentially (if network works) update the store at some point in the future.

The better way to implement is:
1. have some kind of `updateRequired` property in the store (e.g. at the proposals level)
2. Have an action that sets this property to true if new data should be loaded at next chance
3. In `bootApplication` create an interval that checks if something needs to be updated, reloads data, updates store

Reply:
I'm not sure how (3.) would get along with the periodic update of the proposals. I think it's also needed to get changes from other users.

Should I have a `setInterval` each X seconds that updates the proposals if `updateRequired`, and another `setInterval` each Y seconds that updates the proposals regardless?

Reply:
What about this: you have an interval that runs very often but is usually cheap, like every 500 ms. Also an additional time stores when the last network request was triggered. 

- if user requested updated, trigger network request
- if last network request is older than 20 seconds, trigger a new one
- otherwise do nothing since the data we have is good enough